crates/ty_python_semantic/src/types.rs

```diff
@@ -5646,6 +5651,25 @@ impl<'db> TypeVarInstance<'db> {
     }
 }
 
+#[derive(Clone, Debug, Hash, PartialEq, Eq, salsa::Update)]
+pub enum TypeVarVariance {
+    Invariant,
+    Covariant,
+    Contravariant,
+    Bivariant,
+}
+
+impl std::fmt::Display for TypeVarVariance {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Invariant => f.write_str("~"),
+            Self::Covariant => f.write_str("+"),
+            Self::Contravariant => f.write_str("-"),
+            Self::Bivariant => Ok(()),
```

Comment:
This lines up with what the Python interpreter displays for typevars, though the `~` sigil for invariant typevars is an unfortunate conflict with `~` for negation in an intersection!

I've made sure that this display update is in a distinct commit so that it's easy to revert if this part of the PR is controversial.

Reply:
Reverted this; if we decide to do this it will be better in a separate PR as you say